beetsplug/mbsubmit.py

```diff
@@ -57,14 +160,250 @@ def __init__(self):
+                algorithms=self._jwt_algorithm,
+            )
+        except InvalidTokenError as e:
+            self._log.error(f"Invalid token: {str(e)}")
+            raise BadRequest()
+
+        if (
+            "task_key" in payload
+            and payload["task_key"] in self._create_release_tasks
+        ):
+            return self._create_release_tasks[payload["task_key"]]
+        else:
+            self._log.error("task_key does not exist")
+            raise BadRequest()
+
+    def _create_release_add(self):
+        token = request.args.get("token")
+        if token is None:
+            self._log.error("Missing token in request")
+            raise BadRequest()
+
+        task = self._get_task_from_token(token)
+        task.browser_opened = True
+        return render_template("create_release_add.html", task=task)
+
+    def _create_release_complete_add(self):
+        token = request.args.get("token")
+        release_mbid = request.args.get("release_mbid")
+        if token is None or release_mbid is None:
+            self._log.error("Missing token or release_mbid in request")
+            raise BadRequest()
+
+        task = self._get_task_from_token(token)
+        task.result_release_mbid = release_mbid
+        return render_template("create_release_complete_add.html", task=task)
+
+    def _start_server(self) -> bool:
+        if self._server:
+            return True
+
+        if (port := self.create_release_server_port) == 0:
+            # Find a free port for us to use. The OS will select a random available one.
+            # We can't pass 0 to waitress.create_server directly, this won't work when
+            #  using hostnames instead of IP addresses for
+            #  create_release_server_hostname, waitress will then bind to multiple
+            #  sockets, with different ports for each.
+            with socket.socket() as s:
+                s.bind((self.create_release_server_hostname, 0))
+                port = s.getsockname()[1]
+
+        try:
+            self._server = waitress.create_server(
+                self.flask_app,
+                host=self.create_release_server_hostname,
+                port=port,
+            )
+            threading.Thread(target=self._server.run, daemon=True).start()
+            self._server_port = port
+            return True
+        except (PermissionError, ValueError, OSError) as e:
+            self._log.error(
+                f"Failed to start internal web server on "
+                f"{self.create_release_server_hostname}:{port}: {str(e)}"
+            )
+            self._server = None
+            return False
+
+    def _stop_server(self):
+        if self._server:
+            self._server.close()
+            self._server = None
+            self._server_port = None
+
+    def _wait_for_condition(self, condition: Callable):
+        t = threading.current_thread()
+        while not condition():
+            time.sleep(0.5)
```

Comment:
Is this necessary? This is a blocking call on the other thread, so is there any penalty to removing it?